client/extensions/woocommerce/index.js

```diff
@@ -151,15 +155,23 @@ function addStorePage( storePage, storeNavigation ) {
 	} );
 }
 
-function createStoreNavigation( context, next ) {
-	const state = context.store.getState();
-	const selectedSite = getSelectedSite( state );
+function mapStateToProps( state ) {
+	let siteId = getSelectedSiteId( state );
+	if ( ! siteId && ( 1 === getCurrentUserSiteCount( state ) ) ) {
+		siteId = getPrimarySiteId( state );
+	}
+	const site = getSite( state, siteId );
 
+	return {
+		site
+	};
+}
+
+function createStoreNavigation( context, next ) {
 	renderWithReduxStore(
 		React.createElement(
-			StoreSidebar, {
+			connect( mapStateToProps )( StoreSidebar ), {
```

Comment:
It was more the syntax of it. Having it all in one very long line of code. That's all.
When it's all strung together like this, it's easier to miss something while reading it. At least it's that way for me.